sleap/gui/commands.py

```diff
@@ -3470,13 +3481,230 @@ def do_action(context: CommandContext, params: dict):
+
+        # Extract only the selected frames into a new Labels object
+        pruned_labels = labels.extract(
+            inds=[frame_to_index[frame] for frame in valid_frame_indices],
+            copy=True  # Ensures a deep copy of the extracted labels
+        )
+
+        # Remap frame indices in pruned_labels to start from 0 while maintaining spacing
+        frame_offset = frame_range[0]
+        for labeled_frame in pruned_labels.labeled_frames:
+            labeled_frame.frame_idx -= frame_offset
+
+        # Initialize VideoWriter
+        height, width = video.height, video.width
+        fps = params["fps"]
+        writer = VideoWriter.safe_builder(params["filename"], height, width, fps)
+
+        # Conditionally show progress bar
+        show_progress = os.getenv("PYTEST_RUNNING") != "1"
+        if show_progress:
+            app = QtWidgets.QApplication.instance() or QtWidgets.QApplication([])
+            progress = QtWidgets.QProgressDialog("Exporting video...", "Cancel", 0, len(valid_frame_indices))
+            progress.setWindowModality(QtCore.Qt.WindowModal)
+            progress.setValue(0)
+        else:
+            progress = None  # Progress bar disabled during tests
+
+        # Write frames to the video
+        try:
+            for idx, frame_idx in enumerate(valid_frame_indices):
+                if show_progress and progress.wasCanceled():
+                    writer.close()
+                    os.remove(params["filename"])
+                    return
+
+                # Read and process frame
+                try:
+                    frame = video.get_frame(frame_idx)
+                    if frame.ndim == 2:  # Grayscale frames
+                        frame = cv2.cvtColor(frame, cv2.COLOR_GRAY2BGR)
+                    writer.add_frame(frame, bgr=True)
+                except Exception as e:
+                    writer.close()
+                    os.remove(params["filename"])
+                    raise RuntimeError(f"Failed to write frame {frame_idx}: {str(e)}")
+
+                # Update progress
+                if show_progress:
+                    progress.setValue(idx + 1)
+                    QtWidgets.QApplication.processEvents()
+
+        finally:
+            writer.close()
+            if show_progress:
+                progress.setValue(len(valid_frame_indices))  # Complete progress
+
+        # Create a new Video object for the output video
+        new_media_video = MediaVideo(
+            filename=params["filename"], grayscale=video.channels == 1, bgr=True
+        )
+        new_video = Video(backend=new_media_video)
+
+        # Update pruned labels to point to the new video
+        for labeled_frame in pruned_labels.labeled_frames:
+            labeled_frame.video = new_video
+
+        pruned_labels.videos = [new_video]
+
+        # Save the pruned labels
+        labels_filename = params["filename"].replace(".mp4", ".slp")
```

Comment:
Use pathlib for path operations
```suggestion
        # Save the pruned labels
        labels_filename = str(Path(params["filename"]).with_suffix(".slp"))
```